spec/outputs/elasticsearch_spec.rb

```diff
@@ -306,6 +306,56 @@
     end
   end
 
+  describe "http client create" do
+    require "logstash/outputs/elasticsearch"
+    require "elasticsearch"
+
+    let(:options) {
+      {
+        "protocol" => "http",
+        "index" => "my-index",
+        "host" => "localhost",
+        "path" => "some-path"
+      }
+    }
+
+    let(:eso) {LogStash::Outputs::ElasticSearch.new(options)}
+
+    let(:manticore_host) {
```

Comment:
Well, it's a little tricky given that the options are scoped entirely to the register function. To make this work with a zero-arity `http_url` method would require saving the options partway through in an instance var then computing the `http_url`. 

I think, given the code structure we have, the url should probably be computed during options parsing and saved to an `@options` instance var (currently options is method scoped). We could then expose that through an accessor to make all the processed options testable. 

Thoughts?

It might also make sense to do a small refactor and split up `#register` a bit to make this understandable. One to idempotently process the config values and return the options hash, another to build the client. It might be nice for register to look like:

```ruby
attr_reader :options
def register
  #... mutex and other instance var setup
  @options = process_options
  build_client
  start_embedded_elasticsearch if @embedded
end
```